src/StatementProvider.js

```diff
@@ -0,0 +1,73 @@
+( function ( wb, $ ) {
+	'use strict';
+
+	/**
+	 * Combination of a claim, a rank and references.
+	 * @mixin wikibase.datamodel.StatementProvider
+	 *
+	 * @param {wikibase.datamodel.StatementGroupSet|null} [statementGroupSet=new wikibase.datamodel.StatementGroupSet()]
+	 * @license GPL-2.0+
+	 */
+	var SELF = wb.datamodel.StatementProvider = function WbDataModelStatementProvider( statementGroupSet ) {
+		statementGroupSet = statementGroupSet || new wb.datamodel.StatementGroupSet();
+
+		if( !( statementGroupSet instanceof wb.datamodel.StatementGroupSet ) ) {
+			throw new Error( 'Given argument `statementGroupSet` is not instance of `wb.datamodel.StatementGroupSet`' );
+		}
+		this._statementGroupSet = statementGroupSet;
+	};
+
+	/**
+	 * @class wikibase.datamodel.Statement
+	 */
+	$.extend( SELF.prototype, {
+		/**
+		 * @property {wikibase.datamodel.StatementGroupSet}
+		 * @protected
+		 */
+		_statementGroupSet: null,
+
+		/**
+		 * @return {wikibase.datamodel.StatementGroupSet}
+		 */
+		getStatements: function () {
+			return this._statementGroupSet;
+		},
+
+		/**
+		 * @param {wikibase.datamodel.Statement} statement
+		 */
+		addStatement: function ( statement ) {
+			this._statementGroupSet.addStatement( statement );
+		},
+
+		/**
+		 * @param {wikibase.datamodel.Statement} statement
+		 */
+		removeStatement: function ( statement ) {
+			this._statementGroupSet.removeStatement( statement );
+		},
+
+		getStatementByGuid: function (guid) {
```

Comment:
Cannot agree.

When we call `Entity.getStatements()` do we expect to get just this entity statements or all the statements of this entity and all the subentities? I would assume former. And then we need a method like `getThisEntityStatements()`.

Then the whole point of this PR vanishes because what I intended was to make Entity responsible for finding the statement I need.